src/client.js

```diff
@@ -9,17 +9,19 @@ const ENDPOINT = 'https://api.mistral.ai';
  * @return {Promise<void>}
  */
 async function initializeFetch() {
-  if (typeof window === 'undefined' ||
-    typeof globalThis.fetch === 'undefined') {
-    const nodeFetch = await import('node-fetch');
-    fetch = nodeFetch.default;
+  if (typeof window === 'undefined') {
```

Comment:
Some comments:

`window` is also not defined in service workers, as used by Web Extensions, using manifest version 3 

It seems the `isNode` flag is used to determine when to return the response.body directly from the `_request` method, vs implementing an async iterator.

In fact, in newer versions of node, with native fetch you need to do much the same, so `isNode` is a somewhat deceptive name here. It's more node-fetch specific if I'm not mistaken. 

This library should work in:
1. browser in a normal page
2. browser in a service worker
3. an older version of node, without native fetch
4. a newer version of node, with native fetch